shared/AppInsightsCore/src/JavaScriptSDK/EnvUtils.ts

```diff
@@ -354,3 +356,139 @@ export function sendCustomEvent(evtName: string, cfg?: any, customDetails?: any)
+    const questionMarkIndex = strIndexOf(url, "?");
+    if (questionMarkIndex === -1) {
+        return url;
+    }
+    // To build a parameter name until we reach the '=' character
+    // If the parameter name is a one to redact, we will redact the value
+    const baseUrl = strSubstring(url, 0, questionMarkIndex + 1);
+    let queryString = strSubstring(url, questionMarkIndex + 1);
+
+    // Extract fragment if present
+    let fragment = STR_EMPTY;
+    const hashIndex = strIndexOf(queryString, "#");
+    if (hashIndex !== -1) {
+        fragment = strSubstring(queryString, hashIndex);
+        queryString = strSubstring(queryString, 0, hashIndex);
+    }
+
+    // Extract parameters
+    const params: { [key: string]: string } = {};
+    if (queryString && queryString.length) {
+        const pairs = queryString.split("&");
+        for (let i = 0; i < pairs.length; i++) {
+            const pair = pairs[i];
+            if (!pair) {
+                continue;
+            }
+            
+            const equalsIndex = strIndexOf(pair, "=");
+            if (equalsIndex === -1) {
+                params[pair] = null;
+            } else {
+                const paramName = pair.substring(0, equalsIndex);
+                const paramValue = pair.substring(equalsIndex + 1);
+                params[paramName] = paramValue;
+            }
+        }
+    }
+    
+    // Check if any parameters need redaction
+    let anyParamRedacted = false;
+    if (config && config.redactQueryParams) {
+        sensitiveParams = DEFAULT_SENSITIVE_PARAMS.concat(config.redactQueryParams);
+    } else {
+        sensitiveParams = DEFAULT_SENSITIVE_PARAMS;
+    }
+    for (let i = 0; i < sensitiveParams.length; i++) {
+        const sensParam = sensitiveParams[i];
+        if (params[sensParam] !== UNDEFINED_VALUE) {
+            params[sensParam] = STR_REDACTED;
+            anyParamRedacted = true;
+        }
+    }
+    
+    // If no parameters were redacted, return the original URL
+    if (!anyParamRedacted) {
+        return url;
+    }
+
+    const parts: string[] = [];
+    objForEachKey(params, (key, value) => {
+        parts.push(value === null ? key : key + "=" + value);
+    });
+
+    return baseUrl + parts.join("&") + fragment;
+}
+
+/**
+ * Redacts sensitive information from a URL string, including credentials and specific query parameters.
+ * @param input - The URL string to be redacted.
+ * @param config - Configuration object that contain redactUrls setting.
+ * @returns The redacted URL string or the original string if no redaction was needed or possible.
+ */
+export function fieldRedaction(input: string, config: IConfiguration): string {
+    if (!input){
+        return input;
+    }
+    if (input.indexOf(" ") !== -1) {
+        return input; // Checking for URLs with spaces
+    }
+    const isRedactionDisabled = config && config.redactUrls === false;
+    if (isRedactionDisabled) {
+        return input;
+    }
+    try {
```

Comment:
Or maybe (extension of the comment above), add the check here to see if the URL contains "@" or any "recactionKey=" before calling the either function,